Add pagination to organization teams table

vgeorge opened the pull request:
This contributes to #319 by adding pagination to the list of teams on the organization page. It should serve as the pattern for all tables in the UI. Changes added:

- Added pagination component to org teams list
- Added `useFetchList` for fetching table data
- Moved "create org team" and "get org teams" routes to Next.js /api folder
- Started refactoring validation and permissions middleware to Next.js using next-connect

To do:

- [x] Make API return the total count of existing teams for an org
- [x] Add documentation to refactored routes
- [x] Add styles
- [x] Add Cypress test coverage for pagination

cc @LanesGood @batpad

LanesGood commented:
@vgeorge Styles added
![image](https://user-images.githubusercontent.com/12634024/207451752-b4cb115c-5478-4fdd-833b-c25a93a7e423.png)

vgeorge commented:
@LanesGood this looks great, thanks for the change. I've hidden the 'id' column in the Teams table. I don't think it is very useful as OSM user ids. I also made the pagination component disappear if there are no records in the database. Let me know if you prefer a different approach.

My recent changes are:

- Added test coverage for the pagination component
- Added swagger docs for the org team routes (I have a plan to reinstate all docs, which will be ticketed)
- Added LOG_LEVEL environment variable to avoid logging expected errors during testing
- Refactored the base handler

@kamicut this is ready for review. You can see pagination working by running `yarn e2e:dev` and visiting the `organization` spec.

batpad reviewed (approved):
@vgeorge this looks really good to me!
